headless.module

```diff
@@ -84,32 +97,156 @@ function headless_menu() {
  */
 function headless_type($type, $nid) {
   $config = config_get('headless.settings', 'node');
+    printf("type = %s\n", $type);
+  printf("config-type = %s\n",$config[$type]);
+  // check if the JSON output is enabled for the selected type 
   if ($config[$type] != 1) {
     $json_error = ['code' => 404];
     backdrop_json_output($json_error);
     backdrop_exit();
   }
-  $my_json = node_load($nid);
-  backdrop_json_output($my_json);
-  backdrop_exit();
+  // load the requested node and check to see if types match or not
+  $requested_node = node_load($nid);
```

Comment:
we can't do this here, because we don't have an `$nid` passed in via URL this time.

Reply:
hmm.. Maybe I'm missing something or going about this the wrong way.  What I did was leave the existing headless_type() and headless_type_v2() methods alone -- the code above I didn't touch as part of this change -- that was associated with #38.. Git is not one of my strong suits and I'm not sure how to 'fix' things to ensure the PR is not showing stuff that I changed as part of the other PR.. Sorry!!!

Reply:
I forgot to say that I added two new methods below the existing ones -- so there's 4 'type' methods now.  If you'd prefer them merged together I can do that I think..Maybe that would simplify things.. I don't know.